### PR TITLE
Simplified rebase

### PR DESCRIPTION
**Pull request by bpeabody**

### CODE REVIEW

**Comment by bpeabody** [comment]
I was not able to recreate https://github.com/twosigma/git-meta/issues/324, but the codepath through with that NPE could happen no longer exists, so I believe this PR will fix that problem.

**Comment by bpeabody** [comment]
Should also address https://github.com/twosigma/git-meta/issues/321

File: node/lib/util/rebase_util.js [hunk 1]
```diff
@@ -120,193 +108,106 @@ const callFinish = co.wrap(function *(repo, rebase) {
...
- * @param {SubmoduleFetcher}   fetcher
+ * @param {NodeGit.Index} index
+ * @return {Boolean}
  */
-function SubmoduleRebaser(submoduleName, repo, submoduleCommits, getFetcher) {
-    assert.isString(submoduleName);
-    assert.instanceOf(repo, NodeGit.Repository);
-    assert.isObject(submoduleCommits);
-    assert.isFunction(getFetcher);
+const stageAll = co.wrap(function *(index) {
```
Comment (novalis) [inline node/lib/util/rebase_util.js]: Isn't this Index#updateAll?

**Comment by novalis** [comment]
Looks like there are also some test failures.

**Comment by bpeabody** [comment]
Two test failures, one I believe from travis flakiness and one from what appears to be flakiness of one of my tests (I've seen it fail there before). Created https://github.com/twosigma/git-meta/issues/335 to address that.

File: node/lib/util/rebase_util.js (hunk 1 above)
Comment (bpeabody) [inline node/lib/util/rebase_util.js]: You might think so, but `updateAll` doesn't update "all", it updates all paths that match an array of wildcards.

**Comment by bpeabody** [comment]
Addressed feedback and squashed the three simplifications into one commit.

File: node/lib/util/rebase_util.js (hunk 1 above)
Comment (novalis) [inline node/lib/util/rebase_util.js]: But "*" would work, right?

**Comment by bpeabody** [comment]
Ping on this one. I think I've addressed everything; it should address most of our current rebase issues.

**Pull request merged**